Panic in json memory gc

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x9dbf7a]

goroutine 13 [running]:
github.com/folbricht/routedns.(*memoryBackend).startGC.func1(0xaf8a20?)
        /root/go/pkg/mod/github.com/folbricht/routedns@v0.1.21-0.20230909090456-cdc9b5482760/cache-memory.go:136 +0x1a
github.com/folbricht/routedns.(*lruCache).deleteFunc(0xc004d128e0, 0xc00d167e48)
        /root/go/pkg/mod/github.com/folbricht/routedns@v0.1.21-0.20230909090456-cdc9b5482760/lru-cache.go:174 +0x5b
github.com/folbricht/routedns.(*memoryBackend).startGC(0xc004d0f2c0, 0xc000205f00?)
        /root/go/pkg/mod/github.com/folbricht/routedns@v0.1.21-0.20230909090456-cdc9b5482760/cache-memory.go:135 +0x149
created by github.com/folbricht/routedns.NewMemoryBackend in goroutine 1
        /root/go/pkg/mod/github.com/folbricht/routedns@v0.1.21-0.20230909090456-cdc9b5482760/cache-memory.go:45 +0x1d7
```

I believe this is in reference to this 
[L136](https://github.com/folbricht/routedns/blob/cdc9b54827607caa46db85cffee61a287a096b1a/cache-memory.go#L136C11-L136C11)
@folbricht I could use some guidance

--- a/lru-cache.go
+++ b/lru-cache.go
@@ -201,6 +201,10 @@ func (c *lruCache) deserialize(r io.Reader) error {
 		if err := dec.Decode(item); err != nil {
 			return err
 		}
+		// Skip bad (or incompatible) records
+		if item.Key.Question.Name == "" || item.Answer == nil {
+			continue
+		}
 		c.addKey(item.Key, item.Answer)
 	}
 	return nil

